models/job.js

```diff
@@ -0,0 +1,179 @@
+"use strict";
+
+const db = require("../db");
+const { BadRequestError, NotFoundError } = require("../expressError");
+const { sqlForPartialUpdate } = require("../helpers/sql");
+
+/** Related functions for jobs. */
+
+class Job {
+    /** Create a job (from data), update db, return new job data.
+     *
+     * data should be { title, salary, equity, company_handle }
+     *
+     * Returns { id, title, salary, equity, company_handle }
+     *
+     * Throws BadRequestError if job already in database.
+     * */
+
+    static async create({ title, salary, equity, companyHandle }) {
+        const duplicateCheck = await db.query(
+            `SELECT title
+                    FROM jobs
+                    WHERE title=$1 AND company_handle=$2`,
+            [title, companyHandle]);
+
+        if (duplicateCheck.rows[0])
+            throw new BadRequestError(`Duplicate job: ${title} at ${companyHandle}`);
+
+        const result = await db.query(
+            `INSERT INTO jobs
+                        (title, salary, equity, company_handle)
+                    VALUES ($1, $2, $3, $4)
+                    RETURNING id, title, salary, equity, company_handle AS "companyHandle"`,
+            [
+                title,
+                salary,
+                equity,
+                companyHandle
+            ],
+        );
+        const job = result.rows[0];
+
+        return job;
+    }
+
+    /** Find all jobs.
+     * 
+     * Accepts an object with a list of filters as a parameter. Will filter the 
+     * results based on provided filter properties.
+     * 
+     * Can be called without being passed an object, or it can be passed an empty
+     *  object.
+     *
+     * Returns [{ id, title, salary, equity, companyHandle }, ...]
+     * */
+
+    static async findAll(filters) {
+        const jobsRes = await db.query(
+            `SELECT id,
+                    title,
+                    salary,
+                    equity,
+                    company_handle AS "companyHandle"
+           FROM jobs
+           ORDER BY title`);
+
+        let jobs = jobsRes.rows;
+
+        if (filters) {
```

Comment:
Could refactor this into a general `findAll` and a helper function that does the optional filtering.

Reply:
Did this. Works for the job model filtering and the company model filtering.